lib/Command/UpdateSystemAutoresponders.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright Copyright (c) 2024 Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @author Richard Steinmetz <richard@steinmetz.cloud>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\Command;
+
+use DateTimeImmutable;
+use OCA\Mail\Account;
+use OCA\Mail\Service\AccountService;
+use OCA\Mail\Service\OutOfOffice\OutOfOfficeState;
+use OCA\Mail\Service\OutOfOfficeService;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\IUserManager;
+use OCP\User\IAvailabilityCoordinator;
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class UpdateSystemAutoresponders extends Command {
+	private ?IAvailabilityCoordinator $availabilityCoordinator;
+
+	public function __construct(
+		private AccountService $accountService,
+		private ITimeFactory $timeFactory,
+		private IUserManager $userManager,
+		private OutOfOfficeService $outOfOfficeService,
+		private ContainerInterface $container,
+	) {
+		parent::__construct();
+
+		try {
+			$this->availabilityCoordinator = $this->container->get(IAvailabilityCoordinator::class);
+		} catch (ContainerExceptionInterface) {
+			$this->availabilityCoordinator = null;
+		}
+	}
+
+	/**
+	 * @return void
+	 */
+	protected function configure() {
+		$this->setName('mail:repair:system-autoresponders');
+		$this->setDescription('Update sieve scripts of all accounts that follow the system out-of-office period');
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		if ($this->availabilityCoordinator === null) {
+			$output->writeln("<error>This command requires Nextcloud >= 28</error>");
+			return 1;
+		}
+
+		$accounts = $this->accountService->findAll();
```

Comment:
could bloat/exhaust memory on larger installations. two ideas
1) add a second findX that returns a generator and loop over the results one by one without ever building an array that holds all accounts
2) add dedicated method to find only accounts that follow system ooo

Reply:
Done